compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1496,15 +1502,27 @@ public void visit(BLangScope scopeNode) {
         symbolEnter.defineNode(scopeNode.compensationFunction.function, env);
         typeChecker.checkExpr(scopeNode.compensationFunction, env);
         symbolEnter.defineNode(scopeNode, env);
+        BSymbol symbol = symResolver.lookupSymbol(env, names.fromString(scopeNode.name.getValue()), SymTag.SCOPE);
+        //Add to map to be used by parent scopes
+        String uniqueName = getUniqueScopeName(symbol.pkgID, symbol.owner, scopeNode.name.getValue());
+        scopesMap.put(scopeNode.name.getValue(), uniqueName);
```

Comment:
When is this map gets cleaned? Seems like data is accumulating. Seems like this logic won't work with two functions which have a componsate statements.